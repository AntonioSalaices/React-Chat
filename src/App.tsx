import React, {
  BaseSyntheticEvent,
  useEffect,
  useState,
  lazy,
  Suspense,
} from "react";

import { PURPLE } from "@Utils/colors";
import { Events } from "@Constans/eventConstants";
import { ThemeProvider } from "context/ThemeContext.tsx";

import { setTranslationsByUserPreferences } from "./i18n";
import i18n from "./i18n/i18n";

const HomePage = lazy(() => import("./screens/home/home"));
const Layout = lazy(() => import("./components/composed/Layout/Layout"));
const Spinner = lazy(() => import("@Components/basics/Spinner/Spinner"));
const ErrorBoundary = lazy(
  () => import("@Components/composed/ErrorBoundary/ErrorBoundary")
);

const App = (): React.ReactElement => {
  const [isLoaded, setIsLoaded] = useState<boolean>(true);

  useEffect(() => {
    const loadTranslationsInitially = () => {
      setIsLoaded(false);
      setTranslationsByUserPreferences(i18n, navigator.language, setIsLoaded);
    };
    loadTranslationsInitially();
  }, []);

  const handleLanguageChanges = (event: Partial<BaseSyntheticEvent>) => {
    setIsLoaded(false);
    const locale: string =
      event?.currentTarget?.clientInformation?.languages[0];
    setTranslationsByUserPreferences(i18n, locale, setIsLoaded);
  };

  useEffect(() => {
    window.addEventListener(Events.LANGUAGE, handleLanguageChanges);

    return () => {
      window.removeEventListener(Events.LANGUAGE, handleLanguageChanges);
    };
  }, []);

  return (
    <ThemeProvider>
      <ErrorBoundary>
        <Suspense fallback={<Spinner singleColor={PURPLE} />}>
          <Layout>
            {isLoaded ? <HomePage /> : <Spinner singleColor={PURPLE} />}
          </Layout>
        </Suspense>
      </ErrorBoundary>
    </ThemeProvider>
  );
};

export default App;
