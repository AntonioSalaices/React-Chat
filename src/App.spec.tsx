import { render } from '@testing-library/react';
import App from './App.tsx';

describe('<App />', () => {
  test('App component should render correctly', () => {
    const AppComponent = render(<App />);

    expect(AppComponent).toBeDefined();
  });
});
