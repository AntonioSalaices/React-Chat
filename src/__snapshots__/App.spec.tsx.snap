// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> App component should render correctly 1`] = `
<div>
  <div
    class="container"
    id="main-container"
  >
    <div
      class="row justify-center"
    >
      <span
        style="display: inline-block; position: relative; width: 28px; height: 28px; vertical-align: middle; font-size: 0px; animation: __react-md-spinner-animation__root-rotate 1568.235294117647ms linear infinite;"
      >
        <span
          style="box-sizing: border-box; display: block; position: absolute; width: 100%; height: 100%; border-color: #5c217f; white-space: nowrap; opacity: 1; animation-name: __react-md-spinner-animation__fill-unfill-rotate, __react-md-spinner-animation__layer-1-fade-in-out; animation-duration: 5332ms; animation-timing-function: cubic-bezier(.4, 0, .2, 1); animation-iteration-count: infinite;"
        >
          <span
            style="display: inline-block; box-sizing: border-box; position: relative; width: 50%; height: 100%; overflow: hidden;"
          >
            <span
              style="display: inline-block; box-sizing: border-box; position: absolute; top: 0px; border-radius: 50%; bottom: 0px; width: 200%; border-width: 3px; border-style: solid; animation-duration: 1333ms; animation-timing-function: cubic-bezier(.4, 0, .2, 1); animation-iteration-count: infinite; left: 0px; transform: rotate(129deg); animation-name: __react-md-spinner-animation__left-spin; border-color: #5c217f transparent transparent #5c217f;"
            />
          </span>
          <span
            style="display: inline-block; box-sizing: border-box; position: relative; width: 50%; height: 100%; overflow: hidden;"
          >
            <span
              style="display: inline-block; box-sizing: border-box; position: absolute; top: 0px; border-radius: 50%; bottom: 0px; width: 200%; border-width: 3px; border-style: solid; animation-duration: 1333ms; animation-timing-function: cubic-bezier(.4, 0, .2, 1); animation-iteration-count: infinite; left: -100%; transform: rotate(-129deg); animation-name: __react-md-spinner-animation__right-spin; border-color: #5c217f #5c217f transparent transparent;"
            />
          </span>
          <span
            style="display: inline-block; box-sizing: border-box; position: absolute; top: 0px; border-radius: 50%; left: 45%; width: 10%; border-width: 3px; border-top-style: solid;"
          />
        </span>
        <span
          style="box-sizing: border-box; display: block; position: absolute; width: 100%; height: 100%; border-color: #5c217f; white-space: nowrap; opacity: 1; animation-name: __react-md-spinner-animation__fill-unfill-rotate, __react-md-spinner-animation__layer-2-fade-in-out; animation-duration: 5332ms; animation-timing-function: cubic-bezier(.4, 0, .2, 1); animation-iteration-count: infinite;"
        >
          <span
            style="display: inline-block; box-sizing: border-box; position: relative; width: 50%; height: 100%; overflow: hidden;"
          >
            <span
              style="display: inline-block; box-sizing: border-box; position: absolute; top: 0px; border-radius: 50%; bottom: 0px; width: 200%; border-width: 3px; border-style: solid; animation-duration: 1333ms; animation-timing-function: cubic-bezier(.4, 0, .2, 1); animation-iteration-count: infinite; left: 0px; transform: rotate(129deg); animation-name: __react-md-spinner-animation__left-spin; border-color: #5c217f transparent transparent #5c217f;"
            />
          </span>
          <span
            style="display: inline-block; box-sizing: border-box; position: relative; width: 50%; height: 100%; overflow: hidden;"
          >
            <span
              style="display: inline-block; box-sizing: border-box; position: absolute; top: 0px; border-radius: 50%; bottom: 0px; width: 200%; border-width: 3px; border-style: solid; animation-duration: 1333ms; animation-timing-function: cubic-bezier(.4, 0, .2, 1); animation-iteration-count: infinite; left: -100%; transform: rotate(-129deg); animation-name: __react-md-spinner-animation__right-spin; border-color: #5c217f #5c217f transparent transparent;"
            />
          </span>
          <span
            style="display: inline-block; box-sizing: border-box; position: absolute; top: 0px; border-radius: 50%; left: 45%; width: 10%; border-width: 3px; border-top-style: solid;"
          />
        </span>
        <span
          style="box-sizing: border-box; display: block; position: absolute; width: 100%; height: 100%; border-color: #5c217f; white-space: nowrap; opacity: 1; animation-name: __react-md-spinner-animation__fill-unfill-rotate, __react-md-spinner-animation__layer-3-fade-in-out; animation-duration: 5332ms; animation-timing-function: cubic-bezier(.4, 0, .2, 1); animation-iteration-count: infinite;"
        >
          <span
            style="display: inline-block; box-sizing: border-box; position: relative; width: 50%; height: 100%; overflow: hidden;"
          >
            <span
              style="display: inline-block; box-sizing: border-box; position: absolute; top: 0px; border-radius: 50%; bottom: 0px; width: 200%; border-width: 3px; border-style: solid; animation-duration: 1333ms; animation-timing-function: cubic-bezier(.4, 0, .2, 1); animation-iteration-count: infinite; left: 0px; transform: rotate(129deg); animation-name: __react-md-spinner-animation__left-spin; border-color: #5c217f transparent transparent #5c217f;"
            />
          </span>
          <span
            style="display: inline-block; box-sizing: border-box; position: relative; width: 50%; height: 100%; overflow: hidden;"
          >
            <span
              style="display: inline-block; box-sizing: border-box; position: absolute; top: 0px; border-radius: 50%; bottom: 0px; width: 200%; border-width: 3px; border-style: solid; animation-duration: 1333ms; animation-timing-function: cubic-bezier(.4, 0, .2, 1); animation-iteration-count: infinite; left: -100%; transform: rotate(-129deg); animation-name: __react-md-spinner-animation__right-spin; border-color: #5c217f #5c217f transparent transparent;"
            />
          </span>
          <span
            style="display: inline-block; box-sizing: border-box; position: absolute; top: 0px; border-radius: 50%; left: 45%; width: 10%; border-width: 3px; border-top-style: solid;"
          />
        </span>
        <span
          style="box-sizing: border-box; display: block; position: absolute; width: 100%; height: 100%; border-color: #5c217f; white-space: nowrap; opacity: 1; animation-name: __react-md-spinner-animation__fill-unfill-rotate, __react-md-spinner-animation__layer-4-fade-in-out; animation-duration: 5332ms; animation-timing-function: cubic-bezier(.4, 0, .2, 1); animation-iteration-count: infinite;"
        >
          <span
            style="display: inline-block; box-sizing: border-box; position: relative; width: 50%; height: 100%; overflow: hidden;"
          >
            <span
              style="display: inline-block; box-sizing: border-box; position: absolute; top: 0px; border-radius: 50%; bottom: 0px; width: 200%; border-width: 3px; border-style: solid; animation-duration: 1333ms; animation-timing-function: cubic-bezier(.4, 0, .2, 1); animation-iteration-count: infinite; left: 0px; transform: rotate(129deg); animation-name: __react-md-spinner-animation__left-spin; border-color: #5c217f transparent transparent #5c217f;"
            />
          </span>
          <span
            style="display: inline-block; box-sizing: border-box; position: relative; width: 50%; height: 100%; overflow: hidden;"
          >
            <span
              style="display: inline-block; box-sizing: border-box; position: absolute; top: 0px; border-radius: 50%; bottom: 0px; width: 200%; border-width: 3px; border-style: solid; animation-duration: 1333ms; animation-timing-function: cubic-bezier(.4, 0, .2, 1); animation-iteration-count: infinite; left: -100%; transform: rotate(-129deg); animation-name: __react-md-spinner-animation__right-spin; border-color: #5c217f #5c217f transparent transparent;"
            />
          </span>
          <span
            style="display: inline-block; box-sizing: border-box; position: absolute; top: 0px; border-radius: 50%; left: 45%; width: 10%; border-width: 3px; border-top-style: solid;"
          />
        </span>
      </span>
    </div>
  </div>
</div>
`;
